models/gday/R/met2model.GDAY.R

```diff
@@ -0,0 +1,321 @@
+##' @param start_date the start date of the data to be downloaded (will
+##'        only use the year part of the date)
+##' @param end_date the end date of the data to be downloaded (will only use
+##'        the year part of the date)
+##' @param overwrite should existing files be overwritten
+##' @param verbose should the function be very verbose
+met2model.GDAY <- function(in.path, in.prefix, outfolder, start_date,
+                           end_date, ..., overwrite=FALSE,verbose=FALSE){
+
+  ## GDAY driver format (.csv):
+  ## 30min: year (-), doy (-; NB. leap years), hod (-), rainfall (mm 30 min-1),
+  ##        par (umol m-2 s-1), tair (deg C), tsoil (deg C), vpd (kPa),
+  ##        co2 (ppm), ndep (t ha-1 30 min-1), wind (m-2 s-1), press (kPa)
+  ##
+  ## Daily:
+  ## 30min: year (-), doy (-; NB. leap years), tair (deg C),
+  ##        rainfall (mm day-1), tsoil (deg C), tam (deg C), tpm (deg C),
+  ##        tmin (deg C), tmax (deg C), tday (deg C), vpd_am (kPa),
+  ##        vpd_pm (kPa), co2 (ppm), ndep (t ha-1 day-1), wind (m-2 s-1),
+  ##        press (kPa), wind_am (m-2 s-1), wind_pm (m-2 s-1),
+  ##        par_am (umol m-2 s-1), par_pm (umol m-2 s-1)
+
+  SW_2_PAR <- 2.3
+  PA_2_KPA <- 0.001
+  SEC_TO_HFHR <- 60.0 * 30.0
+  K_TO_DEG <- -273.15
+
+  if(!require(PEcAn.utils)) print("install PEcAn.utils")
+
+  start_date <- as.POSIXlt(start_date, tz = "GMT")
+  end_date<- as.POSIXlt(end_date, tz = "GMT")
+  out.file <- paste(in.prefix,
+                    strptime(start_date, "%Y-%m-%d"),
+                    strptime(end_date, "%Y-%m-%d"),
+                    "dat", sep=".")
+  out.file.full <- file.path(outfolder, out.file)
+
+  results <- data.frame(file=c(out.file.full),
+                        host=c(fqdn()),
+                        mimetype=c('text/plain'),
+                        formatname=c('GDAY meteorology'),
+                        startdate=c(start_date),
+                        enddate=c(end_date),
+                        dbfile.name=out.file,
+                        stringsAsFactors=FALSE)
+  print("internal results")
+  print(results)
+
+  if (file.exists(out.file.full) && !overwrite) {
+    logger.debug("File '", out.file.full,
+                 "' already exists, skipping to next file.")
+    return(invisible(results))
+  }
+
+  require(ncdf4)
+  require(lubridate)
+  require(PEcAn.data.atmosphere)
+  #  require(ncdf)
+
+  ## check to see if the outfolder is defined, if not create directory for
+  ## output
+  if (!file.exists(outfolder)){
+    dir.create(outfolder)
+  }
+
+  ## For now setting this to be always true till I figure out how to
+  ## interface with the sub_daily param file. Should detech if met-data
+  ## is coarser than 30-min and swapped to day version?
+  sub_daily = TRUE
+
+  out <- NULL
+
+  ## write the expected header information
+  if (sub_daily) {
+    ounits <- paste("#--,--,--,mm/30min,umol/m2/s,degC,degC,kPa,ppm,"
+                    "t/ha/30min,m/s,kPa", sep="")
```

Comment:
This line is causing the build to fail because there's no comma between these two strings.